lib/watcher.js

```diff
@@ -173,10 +175,68 @@ Watcher.prototype.updateExclusivity = function (file, hasExclusiveTests) {
 	}
 };
 
+Watcher.prototype.trackFailures = function (api) {
```

Comment:
I don't think this tracking behavior belongs in watcher.

Let's enhance `RunStatus` to track both aggregate and per file failures, and give it the ability to track previous runs.

Here is a possible API:

```js
var testRun = api.run([...listofFiles]);

// someTime later
var previousRun = testRun;
testRun = api.run([...listOfFiles], previousRun);
```

Alternatively, maybe just create an "AggregateStatus" class:

```js
var combinedStatus = new AggregateStatus({
  previousRun: RunStatus-instance,
  currentRun: RunStatus-instance,
  files: list-of-files-that-are-being-rerun, // or not being rerun, whichever makes more sense
  deleted: list-of-deleted-tests
});
```